model/src/test/kotlin/ScannerRunTest.kt

```diff
@@ -36,6 +36,164 @@ import org.ossreviewtoolkit.model.utils.clearVcsPath
 import org.ossreviewtoolkit.utils.ort.Environment
 
 class ScannerRunTest : WordSpec({
+    "init" should {
+        "error on duplicate provenance and scanner scan results" {
+            val provenance = RepositoryProvenance(
+                VcsInfo(type = VcsType.GIT, url = "https://github.com/example.git", revision = "revision"),
+                "revision"
+            )
+            val otherProvenance = RepositoryProvenance(
+                VcsInfo(type = VcsType.GIT, url = "https://github.com/example.git", revision = "other_revision"),
+                "other_revision"
+            )
+            val provenances = setOf(
+                ProvenanceResolutionResult(
+                    id = Identifier("maven::example:1.0"),
+                    packageProvenance = provenance
+                ),
+                ProvenanceResolutionResult(
+                    id = Identifier("maven::other_example:1.0"),
+                    packageProvenance = otherProvenance
+                )
+            )
+
+            val scanner = ScannerDetails("scanner", "1.0.0", "configuration")
+            val otherScanner = ScannerDetails("other-scanner", "1.0.0", "configuration")
+
+            // Shared provenance and scanner.
+            val ex = shouldThrow<IllegalArgumentException> {
+                ScannerRun.EMPTY.copy(
+                    provenances = provenances,
+                    scanResults = setOf(
+                        ScanResult(
+                            provenance = provenance,
+                            scanner = scanner,
+                            summary = ScanSummary.EMPTY.copy(
+                                licenseFindings = setOf(
+                                    LicenseFinding("MIT", TextLocation("file1.txt", 1, 1))
+                                )
+                            )
+                        ),
+                        ScanResult(
+                            provenance = provenance,
+                            scanner = scanner,
+                            summary = ScanSummary.EMPTY.copy(
+                                licenseFindings = setOf(
+                                    LicenseFinding("MIT", TextLocation("file2.txt", 1, 1))
+                                )
+                            )
+                        )
+                    )
+                )
+            }
+
+            val expectedMessage = buildString {
+                appendLine("Found multiple scan results for the same provenance and scanner.")
+                appendLine("Scanner:")
+                appendLine(scanner.toYaml())
+                appendLine("Provenance:")
+                append(provenance.toYaml())
+            }.trimEnd()
+
+            ex.message!!.trimEnd() shouldBe expectedMessage
```

Comment:
I hope that the above avoids the need for `!!.trimEnd()` altogether, as we discourage the use of `!!`.